metagraph/src/graph/alignment/aligner_chainer.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef __ALIGNER_CHAINER_HPP__
+#define __ALIGNER_CHAINER_HPP__
+
+#include "alignment.hpp"
+
+
+namespace mtg {
+namespace graph {
+namespace align {
+
+typedef std::vector<Alignment> Chain;
+typedef Alignment::score_t score_t;
+
+// Given sets of forward and reverse-complement seeds, construct and call chains of seeds
+// until terminate() returns true.
+std::pair<size_t, size_t>
+call_seed_chains_both_strands(std::string_view forward,
+                              std::string_view reverse,
+                              const DeBruijnGraph &graph,
```

Comment:
It seems you only need the value of `k` and not the graph itself. Am I correct?

Reply:
In the release-build code, yes. In debug builds, the graph is also used to check that seeds are still valid after merging them.

Reply:
Maybe you can check it outside this function?

Reply:
there are a few places where the graph is only used to access k. I've changed these.